### PR TITLE
Fix definition of URL constructor

### PR DESCRIPTION
**Pull request by ry**
Attempt 2. See #5521 and #5564.

### CODE REVIEW

File: std/path/posix.ts [hunk 1]
```diff
@@ -435,5 +435,5 @@ export function parse(path: string): ParsedPath {
  * are ignored.
  */
 export function fromFileUrl(url: string | URL): string {
-  return new URL(url).pathname;
+  return new URL(url.toString()).pathname;
```
Comment (kitsonk) [inline std/path/posix.ts:438]: Why not?

```suggestion
  return new URL(String(url)).pathname;
```

Comment (nayeemrmn) [reply std/path/posix.ts:438]:
```suggestion
  return (url instanceof URL ? url : new URL(url)).pathname;
```

**Comment by Bidek56** [comment]
Did this PR make into 1.2.0? I am still seeing this error in 1.2.0. Thx

**Comment by kitsonk** [comment]
@Bidek56 have you reloaded your cache?

**Comment by Bidek56** [comment]
@kitsonk I did reload my cache, but this seems to be a pogo package issue, not deno. It fails on this simple code:

```
import React from 'https://dev.jspm.io/react';
import pogo from 'https://deno.land/x/pogo/main.ts';

const server = pogo.server({ port : 3000 });

server.router.get('/', () => {
    return <h1>Hello, world!</h1>;
});

server.start();
```
I will raise this issue with the **pogo** repo. Thx

**Comment by YuriyTigiev** [comment]
deno - 1.2.0 
Source code of the project - https://github.com/badal-khatri-blogs/deno-graphql-mongodb
deno  run --allow-net  --allow-write --allow-read --allow-plugin  --unstable  app.ts

Check file:///C:/Users/yuriy/Documents/Projects/deno/deno-graphql-mongodb/app.ts
error: TS2345 [ERROR]: Argument of type 'string | URL' is not assignable to parameter of type 'string'.
  Type 'URL' is not assignable to type 'string'.
  return new URL(url).pathname
                 ~~~
    at https://deno.land/std@v0.50.0/path/win32.ts:911:18

TS2345 [ERROR]: Argument of type 'string | URL' is not assignable to parameter of type 'string'.       
  Type 'URL' is not assignable to type 'string'.
  return new URL(url).pathname;
                 ~~~
    at https://deno.land/std@v0.50.0/path/posix.ts:433:18

Found 2 errors.

**Comment by kitsonk** [comment]
@YuriyTigiev your dependencies are not type compatible with Deno 1.2.0.  You need to have updated dependencies that are using `std` 0.61.0.  Your dependencies are using 0.50.0 which is incompatible with Deno 1.2.0.

**Comment by YuriyTigiev** [comment]
> @YuriyTigiev your dependencies are not type compatible with Deno 1.2.0. You need to have updated dependencies that are using `std` 0.61.0. Your dependencies are using 0.50.0 which is incompatible with Deno 1.2.0.

How can I update it? The source code from the project doesn't call std library. I can't set a necessary version in the URL.

**Comment by kitsonk** [comment]
@YuriyTigiev `deno info app.ts` should give you a dependency tree and see what code is calling the wrong version of the `std` library and request that the code be updated to be compatible (or revert your version of Deno)

**Comment by YuriyTigiev** [comment]
Nothing new.  I don't see who calls std@0.50 and I can't change it.  


PS C:\Users\yuriy\Documents\Projects\deno\deno-graphql-mongodb> deno info app.ts
Check file:///C:/Users/yuriy/Documents/Projects/deno/deno-graphql-mongodb/app.ts
error: TS2345 [ERROR]: Argument of type 'string | URL' is not assignable to parameter of type 'string'.
  Type 'URL' is not assignable to type 'string'.
  return new URL(url).pathname
                 ~~~
    at https://deno.land/std@v0.50.0/path/win32.ts:911:18

TS2345 [ERROR]: Argument of type 'string | URL' is not assignable to parameter of type 'string'.
  Type 'URL' is not assignable to type 'string'.
  return new URL(url).pathname;
                 ~~~
    at https://deno.land/std@v0.50.0/path/posix.ts:433:18

TS2339 [ERROR]: Property 'openPlugin' does not exist on type 'typeof Deno'. 'Deno.openPlugin' is an unstable API. Did you forget to run with the '--unstable' flag?
  return Deno.openPlugin(localPath);
              ~~~~~~~~~~
    at https://deno.land/x/plugin_prepare@v0.6.0/mod.ts:64:15

Found 3 errors.

**Comment by ebebbington** [comment]
@YuriyTigiev downgrade from deno v1.2.0 and try again, looks like you also need the `--unstable` flag

**Comment by Bidek56** [comment]
@YuriyTigiev 
[deno-graphql-mongodb](https://raw.githubusercontent.com/badal-khatri-blogs/deno-graphql-mongodb/master/shared/mongodb.ts) is importing mongo@v0.7.0 which is old, please ask the owner or do a PR to upgrade to 0.9.0 which has the deno 1.2.0 fix.
From:
`import { MongoClient } from "https://deno.land/x/mongo@v0.7.0/mod.ts";`
To:
`import { MongoClient } from "https://deno.land/x/mongo@v0.9.0/mod.ts";`

**Comment by YuriyTigiev** [comment]
@ebebbington 

PS C:\Users\yuriy\Documents\Projects\deno\deno-graphql-mongodb> deno 
Deno 1.1.3
exit using ctrl+d or close()
>
PS C:\Users\yuriy\Documents\Projects\deno\deno-graphql-mongodb> deno  run --allow-net  --allow-write --allow-read --allow-plugin  --unstable  app.ts
Check file:///C:/Users/yuriy/Documents/Projects/deno/deno-graphql-mongodb/app.ts
error: TS2345 [ERROR]: Argument of type '{ depth: number; sorted: boolean; trailingComma: boolean; compact: boolean; iterableLimit: number; }' is not assignable to parameter of type 'InspectOptions'.
  Object literal may only specify known properties, and 'sorted' does not exist in type 'InspectOptions'.
        sorted: true,
        ~~~~~~~~~~~~
    at https://deno.land/std@0.61.0/testing/asserts.ts:26:9
PS C:\Users\yuriy\Documents\Projects\deno\deno-graphql-mongodb>

**Comment by YuriyTigiev** [comment]
@Bidek56 


> @YuriyTigiev
> [deno-graphql-mongodb](https://raw.githubusercontent.com/badal-khatri-blogs/deno-graphql-mongodb/master/shared/mongodb.ts) is importing mongo@v0.7.0 which is old, please ask the owner or do a PR to upgrade to 0.9.0 which has the deno 1.2.0 fix.
> From:
> `import { MongoClient } from "https://deno.land/x/mongo@v0.7.0/mod.ts";`
> To:
> `import { MongoClient } from "https://deno.land/x/mongo@v0.9.0/mod.ts";`

Does the link below mean to use the latest version of modules?  
import { MongoClient } from "https://deno.land/x/mongo/mod.ts";

**Comment by Bidek56** [comment]
I believe it's the latest version of "https://deno.land/x/mongo/mod.ts" but inside of it version 0.8.0 is specified. 
So not the latest version of all modules.
Please post these questions in the [Discord](https://discord.gg/deno) for better and quicker response.

**Comment by YuriyTigiev** [comment]
The soft and libraries are very unstable  :( 

import { MongoClient } from "https://deno.land/x/mongo@v0.9.0/mod.ts";

PS C:\Users\yuriy\Documents\Projects\deno\deno-graphql-mongodb> deno  run --allow-net  --allow-write --allow-read --allow-plugin  --unstable  app.ts
INFO load deno plugin "deno_mongo" from local "C:\Users\yuriy\Documents\Projects\deno\deno-graphql-mongodb\.deno_plugins\deno_mongo_eba1b6cad1c0aae92825a91f5e45edba.dll"
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: Error("expected value", line: 1, column: 1)', src\lib.rs:75:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
PS C:\Users\yuriy\Documents\Projects\deno\deno-graphql-mongodb>

**Comment by lucacasonato** [comment]
This is an error in Deno Mongo. You have to use v1.2.0 to use deno_mongo@v0.9.0.

**Comment by YuriyTigiev** [comment]
> This is an error in Deno Mongo. You have to use v1.2.0 to use deno_mongo@v0.9.0.

I have provided the link to the project. Could you try to clone and to run it?

**Pull request merged**